### PR TITLE
Add on-fail bbe

### PR DESCRIPTION
**Pull request by pcnfernando**

### CODE REVIEW

File: examples/error-handling-at-single-place/error_handling_in_single_place.bal [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+public function main() {
+    int negResult = getAccountBalance(-1);
+    int invalidResult = getAccountBalance(200);
+
+    transaction {
+        int parsedNum = check parse("12");
+
+        //Parsing a random string will return an error.
+        //This error will be caught within on fail clause
```
Comment (praneesha) [inline examples/error-handling-at-single-place/error_handling_in_single_place.bal:19]:
```suggestion
        //This error will be caught within the `on-fail` clause.
```

File: examples/error-handling-at-single-place/error_handling_in_single_place.bal [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+function parse(string num) returns int|error {
+    return 'int:fromString(num);
+}
+
+function getAccountBalance(int accountID) returns int {
+       do {
+         if (accountID < 0) {
+             // Throw an error with "InvalidAccountID" as the reason if
```
Comment (praneesha) [inline examples/error-handling-at-single-place/error_handling_in_single_place.bal:35]:
```suggestion
             // Throw an error with the `InvalidAccountID` as the reason if
```

File: examples/error-handling-at-single-place/error_handling_in_single_place.bal [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+             //`accountID` is less than zero.
```
Comment (praneesha) [inline examples/error-handling-at-single-place/error_handling_in_single_place.bal:36]:
```suggestion
             //the `accountID` is less than zero.
```

File: examples/error-handling-at-single-place/error_handling_in_single_place.bal [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+             InvalidAccountIDError invalidAccoundIdError
+             = InvalidAccountIDError(INVALID_ACCOUNT_ID, accountID = accountID);
+             fail invalidAccoundIdError;
+         } else if (accountID > 100) {
+             // Throw an error with "AccountNotFound" as the reason if
```
Comment (praneesha) [inline examples/error-handling-at-single-place/error_handling_in_single_place.bal:41]:
```suggestion
             // Throw an error with the `AccountNotFound` as the reason if
```

File: examples/error-handling-at-single-place/error_handling_in_single_place.bal [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+             //`accountID` is greater than hundred.
```
Comment (praneesha) [inline examples/error-handling-at-single-place/error_handling_in_single_place.bal:42]:
```suggestion
             //the `accountID` is greater than hundred.
```

File: examples/error-handling-at-single-place/error_handling_in_single_place.bal [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+             AccountNotFoundError accountNotFoundError
+             = AccountNotFoundError(ACCOUNT_NOT_FOUND, accountID = accountID);
+             fail accountNotFoundError;
+         }
+       //The type of e should be the union of the error types that are
```
Comment (praneesha) [inline examples/error-handling-at-single-place/error_handling_in_single_place.bal:47]:
```suggestion
       //The type of `e` should be the union of the error types that are
```

File: examples/error-handling-at-single-place/error_handling_in_single_place.bal [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+       //thrown by the do statement
```
Comment (praneesha) [inline examples/error-handling-at-single-place/error_handling_in_single_place.bal:48]:
```suggestion
       //thrown by the `do` statement.
```

File: examples/error-handling-at-single-place/error_handling_in_single_place.description [hunk 8]
```diff
@@ -0,0 +1,7 @@
+// On fail catches failures (from fail statement or failure of a check expression) not panics.
```
Comment (praneesha) [inline examples/error-handling-at-single-place/error_handling_in_single_place.description:1]:
```suggestion
// `On-fail` catches failures (from the fail statement or failure of a check expression) but not panics.
```

File: examples/error-handling-at-single-place/error_handling_in_single_place.description [hunk 9]
```diff
@@ -0,0 +1,7 @@
...
+// On fail clause is supported for do, transactions, retry, lock, match, foreach and while statements.
```
Comment (praneesha) [inline examples/error-handling-at-single-place/error_handling_in_single_place.description:2]:
```suggestion
// The `on-fail` clause is supported for `do`, `transactions`, `retry`, `lock`, `match`, `foreach`, and `while` statements.
```

File: examples/error-handling-at-single-place/error_handling_in_single_place.description [hunk 10]
```diff
@@ -0,0 +1,7 @@
...
+// On fail will be executed if the block statement following supported statement fails.
```
Comment (praneesha) [inline examples/error-handling-at-single-place/error_handling_in_single_place.description:3]:
```suggestion
// `On-fail` will be executed if the block statement following the supported statement fails.
```

File: examples/error-handling-at-single-place/error_handling_in_single_place.description [hunk 11]
```diff
@@ -0,0 +1,7 @@
...
+// The type of error caught from on-fail clause will be union of the error types that can be
```
Comment (praneesha) [inline examples/error-handling-at-single-place/error_handling_in_single_place.description:4]:
```suggestion
// The type of error caught from the `on-fail` clause will be the union of the error types that can be
```

**Review by mohanvive — APPROVED** [review]

**Comment by niveathika** [comment]
@pcnfernando Add meta tags file. In addition, batch the review suggestions together and apply from now on.

**Review by praneesha — APPROVED** [review]

**Pull request merged**